segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/SystemJournal.java

```diff
@@ -575,18 +579,19 @@ public void validateAndSaveSnapshot(MetadataTransaction txn) throws Exception {
 
         // Write snapshot
         val snapshotFile = getSystemJournalChunkName(containerId, epoch, 0);
-        currentHandle = chunkStorage.create(snapshotFile).get();
+
         ByteArraySegment bytes;
         try {
             bytes = SYSTEM_SNAPSHOT_SERIALIZER.serialize(systemSnapshot);
         } catch (IOException e) {
             throw new ChunkStorageException(getSystemJournalChunkName(), "Unable to serialize", e);
         }
         synchronized (lock) {
-            // Persist
-            writeToJournal(bytes);
+            currentHandle = chunkStorage.createWithContent(snapshotFile,
```

Comment:
Here you are holding a lock while executing and awaiting an async task. This is a recipe for deadlocks. Is there no other way to do this?

Reply:
This lock is used to serialize writing to journal file. This will be called to record changes to special SLTS system segments only.  
This is a "leaf level" write that only does one thing - it writes to chunk and that's it. So there will be no possibility deadlock here.